kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -166,6 +172,34 @@ showConfig configNode = do
                 $ node'
         else putStrLn' "Invalid node!"
 
+data NodeStates = StuckNode | UnevaluatedNode
+    deriving (Eq, Ord, Show)
+
+showLeafs
+    :: MetaOrObject level
+    => StateT (ReplState level) Simplifier ()
+showLeafs = do
+    Strategy.ExecutionGraph { graph } <- Lens.use lensGraph
+    let nodes = Graph.nodes graph
+    let leafs = filter (\x -> (Graph.outdeg graph x) == 0) nodes
+    let result =
+            groupSort
+                . catMaybes
+                . fmap (f graph)
+                $ leafs
+
+    putStrLn' $ foldr ((<>) . showPair) "" result
+  where
+    f graph node =
```

Comment:
I know I helped with this refactor, but we forgot to rename `f` to something sensible. How about `getNodeState`?